### PR TITLE
[6.0] Disabled SimpleMessage’s property and Message for render in Mailer

### PR DESCRIPTION
**Pull request by cocoeyes02**
This PR is related to [laravel/ideas#1792](https://github.com/laravel/ideas/issues/1792)

# Problem
In class extends Notification, I inserted array ['message' => 'hoge'] at view method's second param. 
Then I executed dd($message) in view blade file, but it’s not result I expected.

## For example
```php
return (new MailMessage())
    ->subject('email.hoge')
    ->view(
        'hoge',
        [
            'message' => 'huga'
        ]
    );
```

hoge.blade.php
```php
dd($message)
```

### Expected
`huga`

### Result
スクリーンショット 2019-08-15 17 45 13

# cause
when using Render method Illuminate\Mail\Mailer, `$data` includes "message" property and `SimpleMessage`’s property.
https://github.com/laravel/framework/blob/5eef06c7b6a21857a41c7eada53303a94b87edad/src/Illuminate/Mail/Mailer.php#L213
https://github.com/laravel/framework/blob/5eef06c7b6a21857a41c7eada53303a94b87edad/src/Illuminate/Notifications/Channels/MailChannel.php#L64

In other words, We can’t insert those keys in array at view method’s second param.
```
message
level
subject
greeting
salutation
introLines
outroLines
actionText
actionUrl
```

I think it’s not intuitive.
it shouldn’t know "message" and SimpleMessage’s property when render tamplate.

# What I changed?
I disabled SimpleMessage’s property and Message for render in src/Illuminate/Mail/Mailer.php.

### CODE REVIEW

**Comment by X-Coder264** [comment]
The documentation already mentions

> Laravel automatically makes the $message variable available to all of your email templates

IMO, rather than doing this (which creates a breaking change for little benefit) the documentation could be updated to better explain that there are reserved words that cannot be used as view parameter variables.

**Comment by cocoeyes02** [comment]
> The documentation already mentions
> 
> > Laravel automatically makes the $message variable available to all of your email templates
> 
> IMO, rather than doing this (which creates a breaking change for little benefit) the documentation could be updated to better explain that there are reserved words that cannot be used as view parameter variables.

Okay, If I add sentense which explain that there are reserved words that cannot be used as view parameter variables, should I write around this?
https://github.com/laravel/docs/blame/6.0/mail.md#L170

**Comment by X-Coder264** [comment]
I'd say yes. And a mention in the notifications docs should probably be made too.
